hyrule/control.hy

```diff
@@ -1,10 +1,54 @@
 (require
   hyrule.macrotools [defmacro/g! defmacro!])
 (import
-  hyrule.collections [prewalk coll?]
+  hyrule.anaphoric [recur-sym-replace]
+  hyrule.collections [prewalk by2s]
+  hyrule.iterables [coll?]
   hyrule.misc [inc])
 
 
+(defmacro branch [tester #* rest]
+  (_branch tester rest))
+
+(defmacro ebranch [tester #* rest]
+  (_branch tester (+ rest
+    (, 'else '(raise (ValueError "ebranch: No branch matched"))))))
+
+(defn _branch [tester rest]
+  (when (% (len rest) 2)
+    (raise (TypeError "each case-form needs a result-form")))
+  (setv it (hy.gensym "branch-it"))
+  `(cond ~@(gfor [case result] (by2s rest) `[
+    ~(if (= case 'else)
+      'True
+      `(do
+        (setv ~it ~case)
```

Comment:
That would lead to evaluating the case form more than once whenever `it` appeared multiple times in the tester, wouldn't it? I should probably have a test to make sure a bug like doesn't make it in (or isn't already there).